### DISCUSSION TITLE
ShortSignatures in BLS

### DISCUSSION THREAD

**Discussion by secure12** [paulmillr/noble-curves#139] [General]
I have been using @noble/curves v1.2.0 to extend the BLS class to BLS12-377. It works quite well. But today, when I wanted to update the package to v1.4.0, there are type errors because ShortSignatures were added and are required in BLS. I don't need signatures in my application, it seems weird that they are compulsory in the class.

Also, the missing functions are just serialisation of signatures (to/from hex/bytes). Why does the short signature implementation need to re-implement the same functionalities in the EC groups? Aren't the signatures group elements themselves?
E.g.
https://github.com/paulmillr/noble-curves/blob/e14e37ae3415b634d1fc7d751ea8b11a59dda35d/src/bls12-381.ts#L1201-L1216
and
https://github.com/paulmillr/noble-curves/blob/e14e37ae3415b634d1fc7d751ea8b11a59dda35d/src/bls12-381.ts#L1152-L1181
are very similar. Will there be an issue if `ShortSignature.fromHex` just use `G1.fromBytes` (subject to refactoring)?

**Comment by paulmillr** [comment]
>I don't need signatures in my application, it seems weird that they are compulsory in the class.

Then you can safely make `ShortSignature.fromHex` empty.

Also, consider contributing bls377 to the repo!
